Stylo again

emilio opened the pull request:
This fixes some bugs in our bindings, the most important one being that we didn't repr(C) opaque types.

r? @bholley

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/10358)

<!-- Reviewable:end -->

emilio committed:
geckolib: regenerate style structs with the latest gecko and bindgen

This solves the clone-large-arrays problem.

emilio committed:
geckolib: Check bindgen status and run tests automatically

emilio committed:
geckolib: Update bindings to repr(C) opaques

bholley commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit e81b57c has been approved by `bholley`

<!-- @bors-servo r=bholley e81b57c8daf00fdbf5d2405b0486927885765795 -->

emilio committed:
geckolib: Add dummy fields in the vtables to supress rustc warnings about zero-sized ctypes

This includes https://github.com/emilio/rust-bindgen/commit/6de91cbb8ddf87f8080ab4a381b4ba7f55ed11bc

emilio commented:
@bors-servo: r- force clean

emilio commented:
@bors-servo: r=bholley

 * Re-approving to include the last commit, which suppresses warnings due to the `rustc` quirk explained in https://github.com/emilio/rust-bindgen/commit/6de91cbb8ddf87f8080ab4a381b4ba7f55ed11bc

bors-servo commented:
:pushpin: Commit 51ebdbd has been approved by `bholley`

<!-- @bors-servo r=bholley 51ebdbd5598ebb6db59175f7645094916b7152f9 -->

bors-servo commented:
:hourglass: Testing commit 51ebdbd with merge 71428b5...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/45), [arm32](http://build.servo.org/builders/arm32/builds/50), [linux-dev](http://build.servo.org/builders/linux-dev/builds/49), [linux-rel](http://build.servo.org/builders/linux-rel/builds/50), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/51), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/49), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/50), [status-appveyor](https://ci.appveyor.com/project/servo/servo/build/1.0.1779)
